fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/rdf/impl/VersionsRdfContext.java

```diff
@@ -97,17 +97,25 @@ public VersionsRdfContext(final FedoraResource resource,
                     return new RdfStream();
                 }
 
-                final Node frozenNode = version.getFrozenNode();
-                final com.hp.hpl.jena.graph.Node versionSubject
-                        = nodeToResource(idTranslator).convert(frozenNode).asNode();
+                final String[] labels = versionHistory.getVersionLabels(version);
+                if (labels.length == 0) {
+                    LOGGER.warn("An unlabeled version for {} was found!  Omitting from version listing!",
+                            subject.getURI());
+                    return new RdfStream();
+                } else if (labels.length > 1) {
+                    LOGGER.info("Multiple version labels found for {}!  Using first label, \"{}\".",
+                            subject.getURI(), labels[0]);
+                }
+                final Node versionSubject
+                        = createProperty(subject + "/fcr:versions/" + labels[0]).asNode();
```

Comment:
Please use the constant: https://github.com/fcrepo4/fcrepo4/blob/master/fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/FedoraJcrTypes.java#L67